src/dali/plugin/pair_converter/csv/kraken.py

```diff
@@ -346,160 +424,91 @@ def find_historical_bars(
 
         # Attempt to load smallest duration
         if self.__cached_pairs.get(base_asset + quote_asset + _MINUTE_IN_MINUTES):
-            self.__logger.debug("Retrieving cached bar for %s, %s at %s", base_asset, quote_asset, epoch_timestamp)
+            plural: str = "s" if all_bars else ""
+            self.__logger.debug("Retrieving cached bar %s for %s, %s at %s", plural, base_asset, quote_asset, epoch_timestamp)
             return self._retrieve_cached_bars(base_asset, quote_asset, epoch_timestamp, all_bars, timespan)
 
-        if self._download_and_chunk(base_asset, quote_asset, all_bars):
+        if self._unzip_and_chunk(base_asset, quote_asset, all_bars):
             return self._retrieve_cached_bars(base_asset, quote_asset, epoch_timestamp, all_bars, timespan)
 
         return None
 
-    def _download_and_chunk(self, base_asset: str, quote_asset: str, all_bars: bool = False) -> bool:
-        base_file: str = f"{base_asset}_OHLCVT.zip"
-
-        self.__logger.info("Attempting to load %s from Kraken Google Drive.", base_file)
-        file_bytes = self._google_file_to_bytes(base_file)
-
-        if file_bytes:
-            with ZipFile(BytesIO(file_bytes)) as zipped_ohlcvt:
-                self.__logger.debug("Files found in zipped file - %s", zipped_ohlcvt.namelist())
-                all_timespans_for_pair: List[str]
-                if all_bars:
-                    all_timespans_for_pair = zipped_ohlcvt.namelist()
-                else:
-                    all_timespans_for_pair = [x for x in zipped_ohlcvt.namelist() if x.startswith(f"{base_asset}{quote_asset}_")]
-                if len(all_timespans_for_pair) == 0:
-                    self.__logger.debug("Market not found in Kraken files. Skipping file read.")
+    def _unzip_and_chunk(self, base_asset: str, quote_asset: str, all_bars: bool = False) -> bool:
+        # This function was called because the trading pair hasn't been chunked yet.
+        # In order to chunk a new trading pair, we need the unified CSV file.
+        if not path.exists(self.__UNIFIED_CSV_FILE) and (self.__force_download or self._prompt_download_confirmation()):
+            self.__download_unified_csv()
+
+        self.__logger.info("Attempting to load %s%s from Kraken Google Drive.", base_asset, quote_asset)
```

Comment:
Shouldn't this log go before the `__download_unified_csv()` call just above?

Reply:
I guess it should say "Attempting to retrieve %s%s pair from the unified Kraken CSV file."